src/main/java/duke/main/Duke.java

```diff
@@ -0,0 +1,160 @@
+
+            for (int i =1;i< str.length;i++) {
+                if (str[i].compareTo("/at") == 0) {
+                    for (int j =1;j<i;j++) {
+                        stuff += str[j] + " ";
+                    }
+                    for (int j =i+1;j<str.length;j++) {
+                        deadline += str[j] + " ";
+                    }
+                    break;
+                }
+            }
+            tasks.add(new Event(stuff, parser.parseDate(deadline)));
+        } else {
+            throw new TaskException("tasks");
+        }
+
+        System.out.println("Got it. I've added this duke.task: ");
+        System.out.println(tasks.get(tasks.size()-1).toString());
+        System.out.printf("Now you have %d tasks in the list.\n", tasks.size());
+    }
+
+    /**
+     * Runs the whole Duke program and processes input
+     * until the 'bye' command is given.
+     */
+    public void run() {
+        ui.welcome();
+        Scanner sc = new Scanner(System.in);
+
+        while (true) {
+            System.out.println(ui.dividerLine());
+            String whole_str = sc.nextLine();
+            String[] str = whole_str.split(" ");
+
+            if (str[0].compareTo("bye") == 0) {
+                ui.bye();
+                break;
+            } else if (str[0].compareTo("list") == 0) {
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < tasks.size(); i++) {
+                    System.out.printf("%d. %s\n", i + 1, tasks.get(i));
+                }
+            } else if (str[0].compareTo("mark") == 0) {
+                int k = Integer.parseInt(str[1]);
+                tasks.get(k - 1).setAsDone();
+                System.out.println("Nice! I've marked this duke.task as done: ");
+                System.out.println(tasks.get(k - 1));
+            } else if (str[0].compareTo("unmark") == 0) {
+                int k = Integer.parseInt(str[1]);
+                tasks.get(k - 1).setAsUndone();
+                System.out.println("OK, I've marked this duke.task as not done yet:");
+                System.out.println(tasks.get(k - 1));
+            } else if (str[0].compareTo("delete") == 0) {
+                int k = Integer.parseInt(str[1]);
+                System.out.println("Noted. I've removed this duke.task: ");
+                System.out.println(tasks.get(tasks.size()-1).toString());
+                tasks.remove(k-1);
+                System.out.printf("Now you have %d tasks in the list.\n", tasks.size());
+            } else if (str[0].compareTo("find") == 0) {
+                String toFind = str[1];
+
+                int count = 1;
+                System.out.println("Here are the matching tasks in your list:");
+                for (Task task: tasks) {
+                    if (task.getDescription().contains(toFind)) {
+                        System.out.printf("%d. %s\n", count, task);
+                    }
+                }
```

Comment:
Comprehensive if-else statements!

However, just a minor suggestion. The anonymous variables declared inside each if-else and the different system println makes it hard to read the code 💫  Perhaps some extra indentation or shifting the code out of this if-else is more appropriate?

Overall, still good work 😄